src/core/crypto/crypto_platform_psa.cpp

```diff
@@ -0,0 +1,795 @@
+/*
+ *    Copyright (c) 2025, The OpenThread Authors.
+ *    All rights reserved.
+ *
+ *    Redistribution and use in source and binary forms, with or without
+ *    modification, are permitted provided that the following conditions are met:
+ *    1. Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *    2. Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *    3. Neither the name of the copyright holder nor the
+ *       names of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written permission.
+ *
+ *    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ *    ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ *    WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ *    DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY
+ *    DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ *    (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ *    ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ *    (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ *    SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ *   This file implements the default/weak Crypto platform APIs using ARM PSA API.
+ */
+
+#include "openthread-core-config.h"
+
+#include <string.h>
+
+#include <mbedtls/asn1.h>
+#include <psa/crypto.h>
+
+#include <openthread/instance.h>
+#include <openthread/platform/crypto.h>
+#include <openthread/platform/entropy.h>
+#include <openthread/platform/memory.h>
+
+#include "common/code_utils.hpp"
+#include "common/debug.hpp"
+#include "common/new.hpp"
+#include "config/crypto.h"
+#include "crypto/ecdsa.hpp"
+#include "crypto/hmac_sha256.hpp"
+#include "crypto/storage.hpp"
+#include "instance/instance.hpp"
+
+using namespace ot;
+using namespace Crypto;
+
+#if OPENTHREAD_CONFIG_CRYPTO_LIB == OPENTHREAD_CONFIG_CRYPTO_LIB_PSA
+
+//---------------------------------------------------------------------------------------------------------------------
+// Default/weak implementation of crypto platform APIs
+
+static otError psaToOtError(psa_status_t aStatus)
+{
+    switch (aStatus)
+    {
+    case PSA_SUCCESS:
+        return kErrorNone;
+    case PSA_ERROR_INVALID_ARGUMENT:
+        return kErrorInvalidArgs;
+    case PSA_ERROR_BUFFER_TOO_SMALL:
+        return kErrorNoBufs;
+    default:
+        return kErrorFailed;
```

Comment:
Should follow single `return` from any function/method (per our style guide)?
There are multiple instances of this  this in this source file.